src/Raven.Client/Documents/Commands/GetDocumentsCommand.cs

```diff
@@ -95,6 +103,18 @@ public override HttpRequestMessage CreateRequest(JsonOperationContext ctx, Serve
                 }
             }
 
+            if (_counters != null)
+            {
+                if (_counters.Length == 0)
+                {
+                    pathBuilder.Append("&counter=").Append(Constants.Counters.All);
```

Comment:
Why does sending empty array means all?

Reply:
If we will have a method `IncludeAllCounters` then we can create a ctor that covers that (bool variable `includeAllCounters`) and in other ctors check if array is not null or empty. Which will make this part of the code more clear I think.

Reply:
Yes, I like that

Reply:
fixed